tests/test_standalone_app.py

```diff
@@ -170,3 +170,100 @@ def test_execute_script(
 
         # Now check that the expected output was added to the end of the output widget text.
         assert actual_output == expected_output
+
+
+@pytest.mark.parametrize(
+    "script, altered_script, operation, selection_range",
+    [
+        (
+            "a = 1\n    b = 2\n c = 3",
+            "    a = 1\n        b = 2\n    c = 3",
+            "indent",
+            None,
+        ),
+        (
+            "a = 1\n    b = 2\n c = 3\n        d = 4\n     e = 5",
+            "a = 1\nb = 2\nc = 3\n    d = 4\n    e = 5",
+            "unindent",
+            None,
+        ),
+        # test adding a comment
+        (
+            "a = 1\n    # b = 2\n c = 3\n        d = 4",
+            "# a = 1\n#     # b = 2\n#  c = 3\n#         d = 4",
+            "comment",
+            None,
+        ),
+        # test removing a comment
+        ("# a = 1\n    # b = 2", "a = 1\n    b = 2", "comment", None),
+        # test comments being added at the correct level of indentation
+        ("    a = 1\n    b = 2", "    # a = 1\n    # b = 2", "comment", None),
+        # test removing character when cursor in indentation
+        ("    a = 1", "a = 1", "delete", (4, 4)),
+        # test removing character when cursor in indentation
+        ("     a = 1", "    a = 1", "delete", (4, 4)),
+        # test removing character when cursor not in indentation
+        ("    a = 1", "    a = ", "delete", (9, 9)),
+        # test adding a new line
+        ("    a = 1", "    a = 1\n    ", "new_line", (9, 9)),
+        # test adding a to a new line to a selection
+        ("    a = 1", "    a\n    ", "new_line", (5, 9)),
+        # test adding a new line after a colon, it should auto indent by 4
+        ("    def test(): ", "    def test(): \n        ", "new_line", (16, 16)),
+        # test adding a new line with the selection before the colon, it should indent to the same level.
+        ("    def test(): ", "    def test(\n    ): ", "new_line", (13, 13)),
+    ],
+)
+def test_block_text_operations(
```

Comment:
I love that you used parametrized for this.

I find the test data to be a bit hard to read however.
Have you looked into using `"""` strings instead? I'm not saying it would actually make it better. I simply find the visual parsing of those `\n`-ended lines difficult.

Reply:
Ahah, I can use my suggestion above, and wrap the whole parameterize block in a 
`# fmt: off/on` comments, so that black leaves it alone.
Usually, I would say it is better to leave black to do its thing, but I think here it is justified to not reformat, as it makes it less clear.